include/zephyr/bluetooth/mesh/dfd_srv.h

```diff
@@ -128,13 +194,74 @@ struct bt_mesh_dfd_srv {
 	struct bt_mesh_blob_cli_inputs inputs;
 
 	struct {
+		bool is_oob;
 		enum bt_mesh_dfd_upload_phase phase;
-		const struct bt_mesh_dfu_slot *slot;
+		struct bt_mesh_dfu_slot *slot;
 		const struct flash_area *area;
 		struct bt_mesh_blob_srv blob;
+		struct {
+			uint8_t uri_len;
+			uint8_t uri[CONFIG_BT_MESH_DFU_URI_MAXLEN];
+			uint16_t current_fwid_len;
+			uint8_t current_fwid[CONFIG_BT_MESH_DFU_FWID_MAXLEN];
+			struct bt_mesh_msg_ctx ctx;
+		} oob;
 	} upload;
+
+	struct {
+		uint8_t *schemes;
+		uint8_t count;
+	} oob_schemes;
 };
 
+/** @brief Call when an OOB check has completed or failed
+ *
+ *  This should be called by the application after an OOB check started by the @c start_oob
+ *  callback has completed or failed. The @p status param should be set to one of the following
+ *  values:
+ *
+ *  * @c BT_MESH_DFD_SUCCESS if the check was succesfull and a new firmware ID was found.
+ *  * @c BT_MESH_DFD_ERR_URI_MALFORMED if the URI is not formatted correctly.
+ *  * @c BT_MESH_DFD_ERR_URI_NOT_SUPPORTED if the URI scheme is not supported by the node.
+ *  * @c BT_MESH_DFD_ERR_URI_UNREACHABLE if the URI can't be reached.
+ *  * @c BT_MESH_DFD_ERR_NEW_FW_NOT_AVAILABLE if the check completes successfully but no new
+ *    firmware is available.
+ *
+ *  If this function returns 0, the application should then download the firmware to the
+ *  slot. If an error code is returned, the application should abort the OOB upload.
+ *
+ *  @param srv      Firmware Distribution Server model instance.
+ *  @param slot     The slot used in the OOB upload.
+ *  @param status   The result of the firmware check.
+ *  @param fwid     If the check was successful and new firmware found, this should point to a
+ *                  buffer containing the new firmware ID to store.
+ *  @param fwid_len The length of the firmware ID pointed to by @p fwid.
+ *
+ *  @return 0 on success, (negative) error code otherwise.
+ */
+int bt_mesh_dfd_srv_oob_check_complete(struct bt_mesh_dfd_srv *srv,
+				       const struct bt_mesh_dfu_slot *slot, int status,
+				       uint8_t *fwid, size_t fwid_len);
+
+/** @brief Call when an OOB store has completed or failed
+ *
+ *  This should be called by the application after an OOB store started by the @c start_oob
```

Comment:
The store happens (or does not) after the application informs the model of the check result, but the application then just starts the upload on its own. It is triggered solely by the `start_oob_upload` callback.

Reply:
I didn't get this.

Reply:
They way I wrote this, "an OOB store" == "an entire process from receiving the OOB start, through the check, and then potential download of image data". I can clear it up a bit if that's unclear, but from the POV of an application, the entire process (start, check, download, done) is "a store operation". There is no separate callback for starting the download after checking, only an API function for telling the model so it can update its state and send out status

Reply:
I can clear it up a bit so it's more understandable. Let me try and then you can see if the new one is more understandable